src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -429,9 +433,16 @@ public Builder addFields(Collection<FieldSpec> fieldSpecs) {
     }
 
     public Builder addField(FieldSpec fieldSpec) {
-      checkArgument(fieldSpec.modifiers.containsAll(kind.implicitFieldModifiers),
-          "%s %s.%s requires modifiers %s", kind, name, fieldSpec.name,
-          kind.implicitFieldModifiers);
+      checkState(kind != Kind.ANNOTATION, "%s %s cannot have fields", kind, name);
+      if (kind == Kind.INTERFACE) {
+        requireExactlyOneOf(EnumSet.of(Modifier.PUBLIC, Modifier.PRIVATE),
+            fieldSpec.modifiers);
+        if (kind == Kind.INTERFACE) {
```

Comment:
That was leftover from when the above `if` had `|| kind == Kind.ANNOTATION`. Fixing.